app/models/like.rb

```diff
@@ -2,5 +2,5 @@ class Like < ApplicationRecord
   belongs_to :likable, polymorphic: true, counter_cache: :like_count
   belongs_to :user
 
-  # validates_presence_of :likeable, :user
+  validates :user_id, uniqueness: { scope: :likable_id }
```

Comment:
I am a little bit concerned about this because it may not allow a user to an article and a comment which both has the same id

Reply:
even i was not sure about this but lets see once after making comment